tests/checkers/rst-yamllint.py

```diff
@@ -0,0 +1,311 @@
+from docutils.io import StringInput
+from docutils.parsers.rst import Directive
+from docutils.parsers.rst.directives import register_directive
+from docutils.parsers.rst.directives import unchanged as directive_param_unchanged
+from docutils.utils import Reporter, SystemMessage
+from yamllint import linter
+from yamllint.config import YamlLintConfig
+from yamllint.linter import PROBLEM_LEVELS
+
+REPORT_LEVELS: set[PROBLEM_LEVELS] = {
+    "warning",
+    "error",
+}
+
+
+class IgnoreDirective(Directive):
+    has_content = True
+
+    def run(self) -> list:
+        return []
+
+
+class CodeBlockDirective(Directive):
+    has_content = True
+    optional_arguments = 1
+
+    # These are all options Sphinx allows for code blocks.
+    # We need to have them here so that docutils successfully parses this extension.
+    option_spec = {
+        "caption": directive_param_unchanged,
+        "class": directive_param_unchanged,
+        "dedent": directive_param_unchanged,
+        "emphasize-lines": directive_param_unchanged,
+        "name": directive_param_unchanged,
+        "force": directive_param_unchanged,
+        "linenos": directive_param_unchanged,
+        "lineno-start": directive_param_unchanged,
+    }
+
+    def run(self) -> list[nodes.literal_block]:
+        code = "\n".join(self.content)
+        literal = nodes.literal_block(code, code)
+        literal["classes"].append("code-block")
+        literal["ansible-code-language"] = self.arguments[0] if self.arguments else None
+        literal["ansible-code-block"] = True
+        literal["ansible-code-lineno"] = self.lineno
+        return [literal]
+
+
+class YamlLintVisitor(nodes.SparseNodeVisitor):
+    def __init__(
+        self,
+        document: nodes.document,
+        path: str,
+        results: list[dict],
+        content: str,
+        yamllint_config: YamlLintConfig,
+    ):
+        super().__init__(document)
+        self.__path = path
+        self.__results = results
+        self.__content_lines = content.splitlines()
+        self.__yamllint_config = yamllint_config
+
+    def visit_system_message(self, node: nodes.system_message) -> None:
+        raise nodes.SkipNode
+
+    def visit_error(self, node: nodes.error) -> None:
+        raise nodes.SkipNode
+
+    def visit_literal_block(self, node: nodes.literal_block) -> None:
+        if "ansible-code-block" not in node.attributes:
+            if node.attributes["classes"]:
+                self.__results.append(
+                    {
+                        "path": self.__path,
+                        "line": node.line or "unknown",
+                        "col": 0,
+                        "message": f"Warning: found unknown literal block! Check for double colons '::'. If that is not the cause, please report this warning. It might indicate a bug in the checker or an unsupported Sphinx directive. Node: {node!r}; attributes: {node.attributes}; content: {node.rawsource!r}",
```

Comment:
```suggestion
                        "message": f"Warning: found unknown literal block! Check for double colons '::' and convert them to code blocks. If that is not the cause, please report this warning. It might indicate a bug in the checker or an unsupported Sphinx directive. Node: {node!r}; attributes: {node.attributes}; content: {node.rawsource!r}",
```
@oraNod ^ is that better?

Reply:
Ah, right. It's implicit code blocks (I guess that's the right way to say it). I forgot you could just have naked `::` double colons. Anyway this looks like a good improvement to me.